core/src/services/oss/core.rs

```diff
@@ -213,28 +213,11 @@ impl OssCore {
     /// # Notes
     ///
     /// before return the user defined metadata, we'll strip the user_metadata_prefix from the key
-    pub fn parse_metadata(
-        &self,
-        path: &str,
-        user_metadata_prefix: &str,
-        headers: &HeaderMap,
-    ) -> Result<Metadata> {
+    pub fn parse_metadata(&self, path: &str, headers: &HeaderMap) -> Result<Metadata> {
         let mut m = parse_into_metadata(path, headers)?;
-
-        let data: HashMap<String, String> = headers
-            .iter()
-            .filter_map(|(key, _)| {
-                key.as_str()
-                    .strip_prefix(user_metadata_prefix)
-                    .and_then(|stripped_key| {
-                        parse_header_to_str(headers, key)
-                            .unwrap_or(None)
-                            .map(|val| (stripped_key.to_string(), val.to_string()))
-                    })
-            })
-            .collect();
-        if !data.is_empty() {
-            m.with_user_metadata(data);
+        let user_meta = parse_prefixed_headers(&headers, X_OSS_META_PREFIX);
```

Comment:
refactored this in the same way of `azblob` and `s3` services, as it was a lot of duplicated code